From 732201ee45008a3f18373ae4526bc07404c3de7b Mon Sep 17 00:00:00 2001
From: Linu Cherian <lcherian@marvell.com>
Date: Tue, 18 Sep 2018 15:36:02 +0530
Subject: [PATCH 0211/1051] soc: octeontx2: Add cgx lmac link status to debugfs

Current link status at cgx lmac level can be read from
"stats" file under cgx*/lmac* dir.

Also, cgx link status prints when PF drivers are not
registered has been removed for consistency.

Signed-off-by: Linu Cherian <lcherian@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/rvu_cgx.c     | 3 ---
 drivers/soc/marvell/octeontx2/rvu_debugfs.c | 9 +++++++++
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2/rvu_cgx.c b/drivers/soc/marvell/octeontx2/rvu_cgx.c
index 54f7a81124ff..727889a7ff1f 100644
--- a/drivers/soc/marvell/octeontx2/rvu_cgx.c
+++ b/drivers/soc/marvell/octeontx2/rvu_cgx.c
@@ -172,9 +172,6 @@ static void cgx_notify_pfs(struct cgx_link_event *event, struct rvu *rvu)
 
 		/* check if notification is enabled */
 		if (!test_bit(pfid, &rvu->pf_notify_bmap)) {
-			dev_info(rvu->dev, "cgx %d: lmac %d Link status %s\n",
-				 event->cgx_id, event->lmac_id,
-				 linfo->link_up ? "UP" : "DOWN");
 			continue;
 		}
 
diff --git a/drivers/soc/marvell/octeontx2/rvu_debugfs.c b/drivers/soc/marvell/octeontx2/rvu_debugfs.c
index 8b2cf8c2d6e7..1bed8d9c7486 100644
--- a/drivers/soc/marvell/octeontx2/rvu_debugfs.c
+++ b/drivers/soc/marvell/octeontx2/rvu_debugfs.c
@@ -617,9 +617,18 @@ static void rvu_dbg_npa_init(struct rvu *rvu)
 
 static int cgx_print_stats(void *cgxd, int lmac_id)
 {
+	struct cgx_link_user_info linfo;
 	int stat = 0, err = 0;
 	u64 tx_stat, rx_stat;
 
+	/* Link status */
+	pr_info("\n=======Link Status======\n\n");
+	err = cgx_get_link_info(cgxd, lmac_id, &linfo);
+	if (err)
+		pr_info("Failed to read link status\n");
+	pr_info("\nLink is %s %d Mbps\n\n",
+		linfo.link_up ? "UP" : "DOWN", linfo.speed);
+
 	/* Rx stats */
 	pr_info("\n=======RX_STATS======\n\n");
 	while (stat < CGX_RX_STATS_COUNT) {
-- 
2.17.1

